## 问题描述

A和B测量同一批样品，A声称自己的仪器精度更高(误差更小)，问该声明是否为真？在什么情况下该假设为假？
- A: [118.5, 120.5, 120.4, 121.8, 121.1]  
- B: [117.2, 119.3, 122.2, 118.3, 120.7]

## 问题解答
进行假设检验：

- **原假设 $H_0$**: $\sigma_A^2 \leq \sigma_B^2$  
- **备择假设 $H_1$**: $\sigma_A^2 > \sigma_B^2$ 

默认总体满足正态分布，均值未知，首先计算两组数据的样本方差


In [28]:
import numpy as np
from scipy import stats

A = np.array([118.5, 120.5, 120.4, 121.8, 121.1])
B = np.array([117.2, 119.3, 122.2, 118.3, 120.7])

n_A = len(A)
n_B = len(B)

#计算样本方差
var_A = np.var(A, ddof=1)
var_B = np.var(B, ddof=1)

print(f"var_A = {var_A:.3f}")
print(f"var_B = {var_B:.3f}")

var_A = 1.513
var_B = 3.873


可以看出样本方差$s^2_A < s^2_B$

利用F分布进行检验

$$
F  = \frac{s^2_A}{s^2_B}
$$

有$F \sim F(4,4)$

利用代码进行计算验证



In [29]:
alpha = 0.05

# 计算 F 统计量及 p 值
F_obs = var_A / var_B
df1, df2 = n_A - 1, n_B - 1

# 右侧 p 值
p_val = 1 - stats.f.cdf(F_obs, df1, df2)

print(f"F 统计量 = s_A² / s_B² = {F_obs:.4f}")
print(f"右侧 p 值 = P(F ≥ {F_obs:.4f}) = {p_val:.4f}")

# 右侧 α 分位数
F_crit_right = stats.f.ppf(1 - alpha, df1, df2)

print(f"临界值 F_{{0.05}}({df1}, {df2}) = {F_crit_right:.4f}")
print(f"拒绝域：F > {F_crit_right:.4f}")



F 统计量 = s_A² / s_B² = 0.3907
右侧 p 值 = P(F ≥ 0.3907) = 0.8076
临界值 F_{0.05}(4, 4) = 6.3882
拒绝域：F > 6.3882


有

$$
p- value = P(F \geq F_{\text{obs}}) = 0.8076 > \alpha = 0.05
$$

且$F_{\text{obs}}$的值也不在拒绝域内

此时我们无法拒绝原假设，但这并不说明原假设叙述准确，而只能说明我们无法接受原假设的反面，因此我们改变假设后重新计算：

- **原假设 $H_0$**: $\sigma_A^2 \geq \sigma_B^2$  
- **备择假设 $H_1$**: $\sigma_A^2 < \sigma_B^2$ 

同理：
$$
F = \frac{s^2_A}{s^2_B}
$$

满足$F \sim F(4,4)$



In [30]:
# 计算 F 统计量及 p 值
F_obs = var_A / var_B
df1, df2 = n_A - 1, n_B - 1

# 左侧 p 值
p_val = stats.f.cdf(F_obs, df1, df2)

print(f"F 统计量 = s_A² / s_B² = {F_obs:.4f}")
print(f"左侧 p 值 = P(F ≤ {F_obs:.4f}) = {p_val:.4f}")

# 左侧 α 分位数
F_crit_right = stats.f.ppf(alpha, df1, df2)

print(f"临界值 F_{{0.05}}({df1}, {df2}) = {F_crit_right:.4f}")
print(f"拒绝域：F < {F_crit_right:.4f}")

F 统计量 = s_A² / s_B² = 0.3907
左侧 p 值 = P(F ≤ 0.3907) = 0.1924
临界值 F_{0.05}(4, 4) = 0.1565
拒绝域：F < 0.1565


此时

$$
p- value = P(F \leq F_{\text{obs}}) = 0.1924 > \alpha = 0.05
$$

$F_{\text{obs}}$的值也不在拒绝域内

同样无法拒绝原假设


## 结果分析

在实际问题中，为了判断A的声明是否为真，我们通常应该将其放置在备择假设中，并利用现有的数据尝试拒绝原假设以确认A声明的正确性。在现有情况下，我们无法接受A声明的正确性，即不能认为“A声称自己的仪器精度更高(误差更小)”是真的。同时相应的，我们也无法认为“B声称自己的仪器精度更高(误差更小)”是真的。通过两种不同的假设方式我们证实了这两点，产生这个结果的原因是样本数量过小，功效函数的检验能力不足。

如果我们要认为“A声称自己的仪器精度更高(误差更小)”这一**假设**为假（即此时将其作为原假设，对应于我们的第一种假设方式），则我们在检验中必须要有足够的证据证明A的方差显著大于B的方差，即在第一种假设方式中拒绝了原假设。

如果我们要认为“A声称自己的仪器精度更高(误差更小)”这一**声明**为假（即此时将其作为备择假设，对应于我们的第二种假设方式），则我们在检验中需要说明我们没有足够的证据证明原假设（即该声明的反面）为假的，则此时我们会倾向认为该声明是假的。